File: aladdinworks8_ui/src/services/ServerService.js
import http from "../http-common"; 

class ServerService {
  getAllServers(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/server/servers`, searchDTO);
  }

  get(serverId) {
    return this.getRequest(`/server/${serverId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/server?field=${matchData}`, null);
  }

  addServer(data) {
    return http.post("/server/addServer", data);
  }

  update(data) {
  	return http.post("/server/updateServer", data);
  }
  
  uploadImage(data,serverId) {
  	return http.postForm("/server/uploadImage/"+serverId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ServerService();

File: src/main/java/com/aladdinworks8/service/ServerService.java
package com.aladdinworks8.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworks8.domain.Server;
import com.aladdinworks8.dto.ServerDTO;
import com.aladdinworks8.dto.ServerSearchDTO;
import com.aladdinworks8.dto.ServerPageDTO;
import com.aladdinworks8.dto.ServerConvertCriteriaDTO;
import com.aladdinworks8.service.GenericService;
import com.aladdinworks8.dto.common.RequestDTO;
import com.aladdinworks8.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ServerService extends GenericService<Server, Integer> {

	List<Server> findAll();

	ResultDTO addServer(ServerDTO serverDTO, RequestDTO requestDTO);

	ResultDTO updateServer(ServerDTO serverDTO, RequestDTO requestDTO);

    Page<Server> getAllServers(Pageable pageable);

    Page<Server> getAllServers(Specification<Server> spec, Pageable pageable);

	ResponseEntity<ServerPageDTO> getServers(ServerSearchDTO serverSearchDTO);
	
	List<ServerDTO> convertServersToServerDTOs(List<Server> servers, ServerConvertCriteriaDTO convertCriteria);

	ServerDTO getServerDTOById(Integer serverId);







}





